### ISSUE TITLE
Dereplicate entries with identical sequence and label

### ISSUE THREAD

**Issue by torognes** [torognes/vsearch#381]
A user suggested adding a new dereplication command that takes both the label and the sequence into account. Only if both the label and the sequence is identical should two entries be merged, otherwise both should be kept.

Please see this VSEARCH Forum discussion for further details:

https://groups.google.com/d/msgid/vsearch-forum/665a1953-fd2f-4aeb-b892-5881674b83fd%40googlegroups.com

**Comment by frederic-mahe** [comment]
An alternative solution for the user could be to use shell tools. Assuming no tabs in sequence headers and sequences on one line (no folding):
```sh
# test: same name, same sequence
printf ">s1\nA\n>s1\nA\n" | paste - - | sort -u | tr "\t" "\n"

# generic code (fasta)
paste - - < input.fas | sort -u | tr "\t" "\n" > output.fas

# generic code (fastq)
paste - - - - < input.fastq | sort -u | tr "\t" "\n" > output.fastq
```
Of course, the code above also gives expected results for the cases "same name, different sequences", "different names, same sequence" and "different names, different sequences".

**Comment by torognes** [comment]
In commit 79b737b9c25246840b6407b1b1e5ad56686bdcdb, I have added the new command called `derep_id` that will dereplicate only when both the identifier in the header (label) and the sequences are identical. Otherwise the command is similar to the `derep_fulllength` command.

**Comment by torognes** [comment]
When dereplicating with reverse complementary matches allowed, vsearch failed to take the headers into account when hashing. This bug was resolved thanks to PR #418 by @wulj2.
